### ISSUE TITLE
Drawable MouseDown event occurs outside of control region after first mouse down event

### ISSUE THREAD

**Issue by commonsuppliz** [picoe/Eto#1854]
Environment: 
       Windows 10 [Version 10.0.19042.685]
       Eto.Forms 2.5.8

## Steps to Reproduce the Problem

1. "dotnet new etoapp -n MyEtoApp"
2. Create Drawable Control as follows
     		var drawable1 = new Drawable();
		drawable1.MouseDown += drawable1_MouseDown;
		drawable1.Size = new Size(300, 300);
		drawable1.BackgroundColor = Colors.Red;

3. Add Following in "drawable1_MouseDown()"
          
                 MessageBox.Show(String.Format("Drawable MouseDown at: {0},{1}", e.Location.X, e.Location.Y));

4. Add drawable1 into Content

			Content = new StackLayout
			{
				Padding = 10,
				Items =
				{
					"Hello World!",
					drawable1 // <====
				}
			};
5. Start App. 
6. Click anywhere out side of drawable1. Nothing Happens. This is ok. 
7. Click inside region of drawable1.
```
---------------------------
Error
---------------------------
Drawable MouseDown at: 19.2,7.2
---------------------------
OK   
---------------------------
```

8.  After 7, Click anywhere outside of MainForm.
```
---------------------------
Error
---------------------------
Drawable MouseDown at: -476,53.6
---------------------------
OK   
---------------------------
```

## Expected Behavior

MouseDown should not fire outside of drawable1

## Actual Behavior

Mousedown occurs outside of MainForm

## Specifications

  - Version:
  - Platform(s):     
       Eto.Forms 2.5.8
  - Operating System(s): 
        Windows 10 [Version 10.0.19042.685]

**Comment by commonsuppliz** [comment]
I tested on Button Control's MouseDown event also, and the result seems ok.

**Comment by commonsuppliz** [comment]
I found this bug when I adding mousedown event for my custom control in my project(https://github.com/commonsuppliz/DiskQuotaCleanup)

**Comment by ZuBsPaCe** [comment]
I can reproduce the issue, but have some relevant information to add.

---

The bug does not occur, if you replace the MessageBox inside of the MouseDown event with this line:

```
Debug.WriteLine(String.Format("Drawable MouseDown at: {0},{1}", e.Location.X, e.Location.Y));
```

In this case, the debug output will only be written, if you click inside of the Drawable.

---

Now, going back to the MessageBox: As you said, it keeps popping up, even if you click outside of the Drawable. You can't even open a Menu by clicking, because the MessageBox will show up first. But, if there is a menu item, which shows an AboutBox for example, you can press the ALT-key and reach the AboutBox by keyboard. Afterwards you can click outside of the Drawable again and everything works fine until you click inside the Drawable.

---

Interpretation: Because of my findings I don't think, the MouseDown event is the direct cause of the bug. I think, the combination of the MouseDown and MessageBox causes the bug. Unfortunately I just started using Eto Forms, so I can't be sure or provide any more detail.

**Comment by commonsuppliz** [comment]
Thank you for confirms the issue. As you report, it seems to be a bug related with MessageBox information. After I upgrade to Eto.Forms 2.7.1, this problem seems to be resolved somehow even using MessageBox as well as Debug.WriteLine(...). MouseDown event won't fire outside the drawable now.  A fix might be inclueded after 2.5.1 to 2.7.1. :-)
Thank you @ZuBsPaCe for checking my report.
